src/fluree/server/handlers/drop.clj

```diff
@@ -0,0 +1,59 @@
+(ns fluree.server.handlers.drop
+  (:require [clojure.core.async :as async :refer [<! go]]
+            [fluree.db.util.core :as util]
+            [fluree.db.util.log :as log]
+            [fluree.server.consensus :as consensus]
+            [fluree.server.consensus.events :as events]
+            [fluree.server.handlers.shared :refer [deref! defhandler]]
+            [fluree.server.watcher :as watcher]))
+
+(set! *warn-on-reflection* true)
+
+(defn monitor-consensus
+  [watcher ledger-id drop-resp-chan]
+  (go
+    (let [drop-resp (<! drop-resp-chan)]
+      (when (util/exception? drop-resp)
+        (log/warn drop-resp "Error dropping ledger")
+        (let [error-event (events/drop-error ledger-id drop-resp)]
+          (watcher/deliver-event watcher ledger-id error-event))))))
+
+(defn queue-consensus
+  [consensus watcher ledger-id]
+  (let [drop-resp-ch (consensus/queue-drop-ledger consensus ledger-id)]
+    (monitor-consensus watcher ledger-id drop-resp-ch)))
+
+(defn monitor-watch
```

Comment:
I think we could generalize the functions in the tranact handler instead of writing new ones. `monitor-consensus` and `monitor-watch` look to be essentially the same as functions in there.